examples/c/task_iter.bpf.c

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: GPL-2.0 OR BSD-3-Clause
+/* Copyright (c) 2020 Facebook */
+#include "vmlinux.h"
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_tracing.h>
+#include <bpf/bpf_core_read.h>
+#include "task_iter.h"
+
+char LICENSE[] SEC("license") = "Dual BSD/GPL";
+
+struct {
+	__uint(type, BPF_MAP_TYPE_PERCPU_ARRAY);
+	__uint(max_entries, 1);
+	__type(key, __u32);
+	__type(value, struct task_iter);
+} task_iter_buf SEC(".maps");
+
+struct task_struct___post514 {
+	unsigned int __state;
+} __attribute__((preserve_access_index));
+
+struct task_struct___pre514 {
+	long state;
+} __attribute__((preserve_access_index));
+
+__u32 get_task_state(void *arg)
+{
+	if (bpf_core_field_exists(((struct task_struct___pre514 *)0)->state)) {
+		struct task_struct___pre514 *task = arg;
+		return task->state;
+	} else {
+		struct task_struct___post514 *task = arg;
+		return task->__state;
+	}
+}
+
+SEC("iter/task")
+int get_tasks(struct bpf_iter__task *ctx)
+{
+	struct seq_file *seq = ctx->meta->seq;
+	struct task_struct *task = ctx->task;
+
+	if (task == (void *)0) { // end
+		return 0;
+	}
+
+	__u32 zero = 0;
```

Comment:
let's set a good example for this pattern, please use static global variable for zero.